src/python/foglamp/configuration_manager.py

```diff
@@ -32,33 +21,106 @@
     sa.Column('value', JSONB),
     sa.Column('ts', sa.types.TIMESTAMP)
 )
-"""Defines the table that data will be inserted into"""
+"""Defines the table that data will be used for CRUD operations"""
 
+_valid_type_strings = ['boolean', 'integer', 'string', 'IPv4', 'IPv6', 'X509 certificate']
 _connection_string = 'postgresql://foglamp:foglamp@localhost:5432/foglamp'
 _logger_name = 'configuration-manager'
+_logger = logging.getLogger(_logger_name)
+
+"""
+General naming convention:
+category(s)
+	category_name - string
+	category_description - string
+	category_val - dict
+		item_name - string (dynamic)
+		item_val - dict
+		    entry_name - string
+		    entry_val - string
+-----------4 fixed entry_name/entry_val pairs-------------------------
+            description_name - string (fixed - 'description')
+                description_val - string (dynamic)
+            type_name - string (fixed - 'type')
+                type_val - string (dynamic - ('boolean', 'integer', 'string', 'IPv4', 'IPv6', 'X509 certificate'))
+            default_name - string (fixed - 'default')
+                default_val - string (dynamic)
+            value_name - string (fixed - 'value')
+                value_val - string (dynamic)
+"""
 
-async def _create_new_category(category_name, category_description, category_json_schema):
+async def _merge_category_vals(category_val_new, category_val_storage):
```

Comment:
Yes. `category_val_prepared = await _merge_category_vals...`  awaits in create_category def